### ISSUE TITLE
Running Dev containers under Orchestration engines provided by Docker Desktop: Swarm Kubernetes

### ISSUE THREAD

**Issue by PavelSosin-320** [microsoft/vscode-remote-release#2899]
Docker Desktop provides 2 orchestration engines by choice: Swarm and Kubernetes. Swarm is the default and only waits for docker swarm init to attach the current node to swarm as a leading node. Docker login also creates the config file with the Swarm as orchestration engine type. In other words, everybody who uses Docker  Dev containers has a valid option to run them not as standalone containers. When running containers under an orchestration engine only sharing of the same deployment among users matters from the resource usage point of view - all users running the same or similar deployments, like based on Alpine share the same resources: Unity FS and Container layers. The addition of similar containers almost doesn't affect resource consumption. I can use several containerized Build systems, toolsets, CLIs, e.t.c. for the about zero cost because of such systems normally reuse the same base image (Alpine) and clean up resources (including disk space!) upon build completion. 
I would like to gain awareness about the Deployment version of my container and resources consumption (from docker inspect ) to prevent **"out-off-the resources" situation** or do **resource limits adjustment** when necessary.
Sometimes very basic "save and restart my container" will allow me to continue my work in the most critical situations. "Upgrade to the recent image version" will free a lot of memory.

**Comment by PavelSosin-320** [comment]
is there some news regarding this issue? Looking at dev container image inspect output I see that the installation of components is executed during container run because it is done as CMD instead of RUN. It will never work! **The container has only 5 seconds to start**. Google promised to some customers that Clusted up and scaling up to the unlimited number of replicas will never take more than 30 sec.
This is max tolerable downtime.
In other words, the container shall be deployed to the cluster ready to serve a user instantly.

**Comment by Chuxel** [comment]
@PavelSosin-320 It would help if you would narrow down your feedback to one thing per issue  and are clear about the scenario you are trying to achieve. (Ideally with repro steps.) 
 Unfortunately it's difficult to understand given the descriptions above.

Near as I can tell it appears to be: 
1. Docker Swarm and / or Kubernetes support - While Docker Compose is supported, Kubernetes is limited to attach support today.  There is also not specific support for Swarm.  See #12.
2. Possibly performance of startup - We are not replicating containers nor supporting scale up/down nor providing a mechanism for production deployments, so I'm unclear how your Google comment applies here. All components are injected dynamically so that specific builds are not required to develop inside them. Once a container is running this is not needed again.
3. Something about the CMD override - This is only for Dockerfile support when `overrideCommand` is not set to `false` and is provided since many containers images exit immediately.  It does not apply to Docker Compose at all.
